docarray/index/backends/milvus.py

```diff
@@ -0,0 +1,328 @@
+                        name="doc_id" if column_name == "id" else column_name,
+                        dtype=DataType.VARCHAR if column_name == "id" else info.db_type,
+                        is_primary=False,
+                        **(
+                            {'max_length': 256}
+                            if info.db_type == DataType.VARCHAR or column_name == "id"
+                            else {}
+                        ),
+                        **(
+                            {'dim': info.n_dim}
+                            if info.db_type == DataType.FLOAT_VECTOR
+                            else {}
+                        ),
+                    )
+                    for column_name, info in self._column_infos.items()
+                ]
+            )
+            self._logger.info("Collection has been created")
+            return Collection(
+                name=self._db_config.collection_name,
+                schema=CollectionSchema(
+                    fields=fields,
+                    description="Collection Schema",
+                ),
+                using='default',
+            )
+
+        return Collection(self._db_config.collection_name)
+
+    def _create_collection_name(self):
+        if self._db_config.collection_name is None:
+            id = uuid.uuid4().hex
+            self._db_config.collection_name = f"{self.__class__.__name__}__" + id
+
+        self._db_config.collection_name = ''.join(
+            re.findall('[a-zA-Z0-9_]', self._db_config.collection_name)
+        )
+
+    def _validate_columns(self):
+        """
+        Validates if the DataFrame contains at least one vector column
+        (Milvus' requirement) and checks that each vector column has
+        dimension information specified.
+        """
+        vector_columns_exist = any(
+            safe_issubclass(info.docarray_type, AbstractTensor)
+            for info in self._column_infos.values()
+        )
+
+        if not vector_columns_exist:
+            raise ValueError(
+                "No vector columns found. Ensure that at least one column is of a vector type"
+            )
+
+        for column_name, info in self._column_infos.items():
+            if info.n_dim is None and safe_issubclass(
+                info.docarray_type, AbstractTensor
+            ):
+                raise ValueError(
+                    f"Dimension information is missing for column '{column_name}' which is vector type"
+                )
+
+    def _build_index(self):
+        index = {
+            "index_type": self._db_config.index_type,
+            "metric_type": self._db_config.index_metric,
+            "params": self._db_config.index_params,
+        }
+
+        self._collection.create_index(self._db_config.index_name, index)
+        self._logger.info(
+            f"Index '{self._db_config.index_name}' has been successfully created"
+        )
+
+    def index(self, docs: Union[BaseDoc, Sequence[BaseDoc]], **kwargs):
```

Comment:
What is the reason for implementing `index()` instead of implementing `_index()` and utilizing `index()` from the base class?

Reply:
In my new implementation, the `index()` function would also do the serialization (there's a column for serialized data), so the index function need to be utilized